modules/programs/hexchat.nix

```diff
@@ -0,0 +1,262 @@
+            };
+            commands = mkOption {
+              type = nullOr (listOf str);
+              description = "Commands to be executed on connecting to server.";
+              example = ''[ "ECHO Greetings fellow Nixer! ]'';
+              default = [ ];
+            };
+            loginMethod = mkOption {
+              type = with lib.types;
+                nullOr (types.enum [
+                  "nickServMsg"
+                  "nickServ"
+                  "challengeAuth"
+                  "sasl"
+                  "serverPassword"
+                  "saslExternal"
+                  "customCommands"
+                ]);
+              description = ''
+                null => Default
+                "nickServMsg" (1) => NickServ (/MSG NickServ + password)
+                "nickServ" (2) => NickServ (/NICKSERV + password)
+                "challengeAuth" (4) => Challenge Auth (username + password)
+                "sasl" (6) => SASL (username + password)
+                "serverPassword" (7) => Server password (/PASS password)
+                "saslExternal" (10) => SASL EXTERNAL (cert)
+                "customCommands" (9) => Custom... (connect commands)
+              '';
+              default = null;
+            };
+            nickname = mkOption {
+              type = str;
+              default = "";
+              description = "Primary nickname";
+            };
+            nickname2 = mkOption {
+              type = str;
+              default = "";
+              description = "Secondary nickname";
+            };
+            options = let
+              channelOptions = submodule {
+                options = {
+                  autoconnect = mkOption {
+                    type = nullOr bool;
+                    description = "Autoconnect to network";
+                    default = false;
+                  };
+                  connect_to_selected_server_only = mkOption {
+                    type = nullOr bool;
+                    description = "Connect to selected server only";
+                    default = true;
+                  };
+                  bypass_proxy = mkOption {
+                    type = nullOr bool;
+                    description = "Bypass proxy";
+                    default = true;
+                  };
+                  use_ssl_for_all_servers = mkOption {
+                    type = nullOr bool;
+                    description = "Use SSL for all servers";
+                    default = false;
+                  };
+                  accept_invalid_ssl_certificates = mkOption {
+                    type = nullOr bool;
+                    description = "Accept invalid SSL certificates";
+                    default = false;
+                  };
+                  use_global_user_information = mkOption {
+                    type = nullOr bool;
+                    description = "Use global user information";
+                    default = false;
```

Comment:
camelCase.

Reply:
BTW, is this the only available options? If not, I would rather prefer you to not have this type and set channel options to be a attrset.

Reply:
This options list is exhaustive. Is "attrset" same as "attrs"? A free-form attribute set?
Manual does not include "attrset":
https://nixos.org/manual/nixos/stable/index.html#sec-option-types
Can you suggestion how this should be implemented? I'm uncertain of how I should go with it.

Reply:
I was thinking of defining no options here and just converting the attrset the user defines to the resulting configuration. But if the list is exhaustive maybe we don't need it.

Reply:
LGTM.